OmgvaPOS/src/DiscountManagement/Controller/DiscountController.cs

```diff
@@ -0,0 +1,177 @@
+        [ProducesResponseType(StatusCodes.Status204NoContent)]
+        [ProducesResponseType(StatusCodes.Status400BadRequest)]
+        [ProducesResponseType(StatusCodes.Status401Unauthorized)]
+        [ProducesResponseType(StatusCodes.Status403Forbidden)]
+        [ProducesResponseType(StatusCodes.Status404NotFound)]
+        [ProducesResponseType(StatusCodes.Status409Conflict)]
+        [ProducesResponseType(StatusCodes.Status500InternalServerError)]
+        public IActionResult UpdateDiscountValidUntil([FromBody] DateTime newValidUntil, long id) {
+            if (!JwtTokenHandler.CanManageBusiness(HttpContext.Request.Headers.Authorization!, _discountService.GetDiscountNoException(id).BusinessId))
+                return Forbid();
+            // there is really nothing else you can update in a discount
+            // since we care about historical data:
+            // changing discount% or discount type would require creating a new discount all together
+            try {
+                _discountService.UpdateDiscountValidUntil(id, newValidUntil);
+                return NoContent();
+            }
+            catch (Exception ex) {
+                _logger.LogError(ex, "An unexpected internal server error occured while updating the discount.");
+                return StatusCode(StatusCodes.Status500InternalServerError, ex.Message);
+            }
+        }
+
+        [HttpDelete("{id}")]
+        [Authorize(Roles = "Admin,Owner")]
+        [ProducesResponseType(StatusCodes.Status200OK)]
+        [ProducesResponseType(StatusCodes.Status401Unauthorized)]
+        [ProducesResponseType(StatusCodes.Status404NotFound)]
+        [ProducesResponseType(StatusCodes.Status500InternalServerError)]
+        public IActionResult ArchiveDiscount(long id) {
+            if (!JwtTokenHandler.CanManageBusiness(HttpContext.Request.Headers.Authorization!, _discountService.GetDiscountNoException(id).BusinessId))
+                return Forbid();
+
+            // essentially update IsArchived to true
+            try {
+                _discountService.ArchiveDiscount(id);
+                return NoContent();
+            }
+            catch (Exception ex) {
+                _logger.LogError(ex, "An unexpected internal server error occured while updating the discount.");
+                return StatusCode(StatusCodes.Status500InternalServerError, ex.Message);
+            }
+        }
+
+        [HttpPost("{discountId}/item/{itemId}")]
+        [Authorize(Roles = "Admin,Owner,Employee")]
+        [ProducesResponseType(StatusCodes.Status200OK)]
+        [ProducesResponseType(StatusCodes.Status401Unauthorized)]
+        [ProducesResponseType(StatusCodes.Status404NotFound)]
+        [ProducesResponseType(StatusCodes.Status500InternalServerError)]
+        //TODO: does not work error in ItemRepository UpdateItem(Item item)
+        public IActionResult UpdateDiscountOfItem(long discountId, long itemId) {
```

Comment:
Gal pakeisti apply?

Reply:
Same endpoint is used for adding and removing discounts from an item.
This is done because in the frontend, discount is applied to items by clicking a checkbox on and off next to an item.